src/sys/unix/kqueue.rs

```diff
@@ -74,7 +74,9 @@ impl Selector {
     ) -> io::Result<bool> {
         let timeout = timeout.map(|to| libc::timespec {
             tv_sec: cmp::min(to.as_secs(), time_t::max_value() as u64) as time_t,
-            tv_nsec: libc::c_long::from(to.subsec_nanos()),
+            // On some platforms C's long is i32, for those we need to ensure
+            // `subsec_nanos` doesn't overflow.
+            tv_nsec: cmp::min(to.subsec_nanos(), libc::c_long::max_value() as u32) as libc::c_long,
```

Comment:
`libc::c_long::max_value() as u32` will overflow on many platforms.  How about `i32::max_value() as u32` instead?

Reply:
I was wondering this myself as well (https://github.com/tokio-rs/mio/pull/948#discussion_r285062700), I think `i32::max_values` would do the trick.